### DISCUSSION TITLE
Efficiency improvement

### DISCUSSION THREAD

**Discussion by Nickdejonghe** [serkri/SmartEVSE-3#127] [Ideas]
It would be great if the smartEVSE Solar MAX Imported current would ONLY be applied to reach the smartEVSE Current MIN value. Today the allowed Imported current is always applied until the smartEVSE current MAX is reached. Si if I don't want to spill net power I need to lower the max allowed charging current, but then I loose solar power if the sun breaks through...
I don't think it's difficult to implement, but unfortunately I'm not a programmer :-(

**Comment by dingo35** [comment]
If you set Max Imported Current at 0, StartCurrent at -6, and StopTimer at 60 minutes (or as long as you want), you are very close to the objective you are trying to reach, arent you?

**Reply by Nickdejonghe** [reply]
Yes and no, in that case you need to wait until the surplus solar power exceeds 6A. With smaller and/or east+west oriented solar panels and/or shadow situations the power builds up slower in the morning and keeps climbing till late afternoon. It would be great to be able to start charging as soon as 50% can be delivered by solar, and stop adding netpower if solar production becomes sufficient. I think it would be a really nice add-on for quite some situations.

**Comment by dingo35** [comment]
Then set your StartCurrent at -3?!

**Comment by Nickdejonghe** [comment]
ok, I didn't know that it would trigger charging at 6A with imported current set to zero, even if solar surplus is only 3A. 
In combination with the timer I can reach my goals close enough. Thank you for the hint!

**Reply by dingo35** [reply]
Pleasure!
